src/app.js

```diff
@@ -6,8 +6,318 @@ import _ from 'underscore';
+const reportStatus = function reportStatus(status, message) {
+  console.log(`Reporting ${ status } message: ${ message}`);
+  const generatedHTML = statusMessageTemplate({
+    status: status,
+    message: message,
+  });
+
+  $('#status-messages ul').append(generatedHTML);
+  $('#status-messages').show();
+};
+
+const filterTrips = function filterTrips(trips) {
+  let filter = $('#filter-value').val().toLowerCase().trim();
+  let filterType = $('#filter-type').val();
+
+  // alert(filterType); // way to test that the input is what we think it will be
+  if (filter === "") {
+    return trips;
+  }
+
+  // want to pass in collection.models -- want to pass in an array of models
+  let results = _.filter(tripsList.models, function(trip) {
+    if (filterType === 'name' || filterType === 'category' || filterType === 'continent') {
+      if (trip.get(filterType).toLowerCase().includes(filter)) {      // string filter
+        return true;
+      }
+    } else {
+      if (trip.get(filterType) <= parseInt(filter)) {   // integer filter
+        return true;
+      }
+    }
+
+    return false;
+  });
+
+  return results;
+}
+
+/////////////////RENDER ALL TRIPS //////////////////
+const renderTrips = function renderTrips(list) {
+  let trips = filterTrips(list);
+
+  $('#trips-table').show();
+  const tripTableElement = $('#trips-list');
+  tripTableElement.html('');
+
+  trips.forEach((trip) => {
+    const generatedHTML = tripsTemplate(trip.attributes);
+    tripTableElement.append(generatedHTML);
+  });
+
+  $('.see-trips-button').hide();
+};
+
+//////////// RENDER INDIVIDUAL TRIPS /////////////////
+const renderTrip = function renderTrip(trip) {
+  const tripElement = $('#trip-detail');
+
+  const generatedHTML = $(tripTemplate(trip.attributes));
+  generatedHTML.on('click', '.button', addReservationHandler);
+  tripElement.html(generatedHTML);
+};
+
+////////////// ADD TRIP HANDLER //////////////////////
+const addTripHandler = function(event) {
+  event.preventDefault();
+
+  const tripData = {};
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+
+    if (field === "continent") {
+      const selectElement = $(`#add-trip-form option[name="${ field }"]`);
+      let selectValue = selectElement.val();
+
+      tripData[field] = selectValue;
+      selectElement.val('');
+
+    } else {
+      const inputElement = $(`#add-trip-form input[name="${ field }"]`);
+      let value = inputElement.val();
+
+      const numArray = ["weeks", "cost"];
+
+      if (numArray.includes(field)){
+        value = Number(value);
+      }
+
+      tripData[field] = value;
+      inputElement.val('');
+    }
+  });
+
+  const trip = new Trip(tripData);
+
+  if (!trip.isValid()) {
+    handleValidationErrors(trip.validationError);
+    return;
+  }
+
+  trip.save({}, {
+    success: (model, response) => {
+      tripsList.add(model);
+      reportStatus('success', 'Successfully add a trip!');
```

Comment:
I would clear the form input fields here.  Otherwise if they have validation errors they have to reenter the data again.